### PR TITLE
Refactored CMakeServerClient.sendRequest() to improve consistency

### PR DESCRIPTION
**Pull request by Zingam**
This PR proposes to make CMakeServerClient.sendRequest() private, more readable and its use in line with other calls.

### CODE REVIEW

**Comment by bobbrow** [comment]
@Zingam thanks for all of the PR's you've started!  It's great to see you getting engaged!  I want to apologize in advance that we haven't reviewed them yet, but we do plan to do so in the coming weeks.  Our priority was to get 1.2.3 out first, so I've been tagging these with 1.3.0 which is the milestone we plan to include these in.

File: src/drivers/cms-client.ts [hunk 1]
```diff
@@ -467,43 +467,63 @@ export class CMakeServerClient {
...
     }
     this._pipe.write('\n[== "CMake Server" ==[\n');
-    this._pipe.write(msg);
+    this._pipe.write(jsonMessage);
     this._pipe.write('\n]== "CMake Server" ==]\n');
-    return pr;
+    return promise;
   }
 
+  /**
```
Comment (Zingam) [inline src/drivers/cms-client.ts]: I propose in a separate commit to prepend a `request` or `do` (e.g. `requestConfigure`) to these methods to better express their purpose.

**Comment by Zingam** [comment]
@bobbrow Thank you. I hope my PRs are inline with your vision for the extension. I used Visual Studio's CMake integration since its introduction and I reported a ton of issues but sadly it was closed source. I hope to be able to contribute a few features available in Visual Studio 2019 that are still missing in this extension like 'install and debug', which seems will be much more difficult than I have anticipated.

**Comment by Zingam** [comment]
I also propose to prepend a `request` or `do` (e.g. `requestCompute`) to these methods to better express their purpose in another commit:
  
- setGlobalSettings
-  getCMakeCacheContent()
-  getGlobalSettings
-  configure
-  compute
-  codemodel
-  codemodel
-  cmakeInputs

**Comment by bobbrow** [comment]
> I hope my PRs are inline with your vision for the extension.

I would say that pretty much anything that has an open issue that we've responded to is fair game.  If you are about to start on a large feature/issue, I would recommend you check in with us first in case we have specific goals around it.  That will help to avoid any frustration if our ideas around the implementation don't align at first.  I hope to draft a roadmap with our priorities soon.

**Pull request merged**